src/pyedb/dotnet/edb_core/edb_data/hfss_pi_simulation_setup_data.py

```diff
@@ -0,0 +1,449 @@
+
+    @min_void_area_to_mesh.setter
+    def min_void_area_to_mesh(self, value):
+        self._simulation_settings.MinVoidAreaToMesh = self._pedb.edb_value(value).ToString()
+
+    @property
+    def model_type(self):
+        """Model Type setting
+
+        Return
+        ------
+            int
+         model type, ``0``=RDL, ``1``=Package, ``2``=PCB
+
+        """
+        return self._simulation_settings.ModelType
+
+    @model_type.setter
+    def model_type(self, value):
+        if isinstance(value, int):
+            self._simulation_settings.ModelType = value
+
+    @property
+    def perform_erc(self):
+        """Perform ERC
+
+        Return
+        ------
+            bool
+
+        """
+        return self._simulation_settings.PerformERC
+
+    @perform_erc.setter
+    def perform_erc(self, value):
+        if isinstance(value, bool):
+            self._simulation_settings.PerformERC = value
+        else:
+            self.logger.error(f"HFSSPI setup perform_erc setter input must be a boolean. Provided value {value}")
+
+    @property
+    def pi_slider_pos(self):
+        """The Simulation Preference Slider setting
+
+        Return
+        ------
+            int
+        ``0``= balanced, ``1``=Accuracy.
+
+        """
+        return self._simulation_settings.PISliderPos
+
+    @pi_slider_pos.setter
+    def pi_slider_pos(self, value):
+        if isinstance(value, int):
+            self._simulation_settings.PISliderPos = value
+        else:
+            self.logger.error(f"HFSSPI setup pi_slider_pos setter input must be a integer. Provided value {value}")
+
+    @property
+    def rms_surface_roughness(self):
+        """RMS Surface Roughness setting
+
+        Return
+        ------
+            str
+        """
+        return self._simulation_settings.RMSSurfaceRoughness
+
+    @rms_surface_roughness.setter
+    def rms_surface_roughness(self, value):
+        self._simulation_settings.RMSSurfaceRoughness = self._pedb.edb_value(value).ToString()
+
+    @property
+    def signal_nets_conductor_modeling(self):
+        """Conductor Modeling
+
+        Return
+        ------
+            str
+        ``"MeshInside"`` or ``"ImpedanceBoundary"``.
+
+        """
+        return self._simulation_settings.SignalNetsConductorModeling
+
+    @signal_nets_conductor_modeling.setter
+    def signal_nets_conductor_modeling(self, value):
+        if isinstance(value, str):
+            self._simulation_settings.SignalNetsConductorModeling = value
+
+    @property
+    def signal_nets_error_tolerance(self):
+        """Error Tolerance
+
+        Return
+            str
+        Value between 0.02 and 1.
+        """
+        return self._simulation_settings.SignalNetsErrorTolerance
+
+    @signal_nets_error_tolerance.setter
+    def signal_nets_error_tolerance(self, value):
+        self._simulation_settings.SignalNetsErrorTolerance = self._pedb.edb_value(value).ToString()
```

Comment:
Add check on value (cf getter description) ?

Reply:
No need because EdbValue class handle and we convert everytime to str.